### ISSUE TITLE
Drupal.sanitizeAjaxUrl is not a function

### ISSUE THREAD

**Issue by paul121** [farmOS/farmOS#352]
> Uncaught TypeError: Drupal.sanitizeAjaxUrl is not a function

Just noticing this, but I cannot add or edit areas. Additionally cannot edit geometries on a log via the map. The issue seems to be related to loading an instance of the map with edit controls? But not necessarily the `edit` behavior?

Testing on test.farmos.net, multiple versions + OS for Firefox, Chrome and Safari desktop.

Looking closer... all AJAX is broken.. can't auto-fill an assets name... same message in console.

From the latest Drupal security advisory: https://www.drupal.org/sa-core-2020-007 :
> Drupal 7 sites should also pass such URLs through the new Drupal.sanitizeAjaxUrl() function.

> If you were previously relying on Drupal's AJAX API to perform trusted JSONP requests, you'll either need to override the AJAX options to set "jsonp: true", or you'll need to use the jQuery AJAX API directly.

> If you are using jQuery's AJAX API for user-provided URLs in a contrib or custom module, you should review your code and set "jsonp: false" where this is appropriate.

Not sure which of these applies to us.... but seems likely the problem is somewhere there...

![Screenshot_2020-09-18_17-21-51](https://user-images.githubusercontent.com/3116995/93654827-a5e73f00-f9d4-11ea-9a71-6a3e5e5e1368.png)

**Comment by paul121** [comment]
Just spun up a fresh install from the last farmos:dev image (with Drupal 7.73) and I'm not seeing this issue. Perhaps its related to the upgrade from 7.72 to 7.73?

**Comment by mstenta** [comment]
Thanks for the heads up @paul121 - I looked into this and figured out what the problem was. I had patched all the farmOS instances on Farmier, but did not clear the caches afterwards. The patch adds a new function `Drupal.sanitizeAjaxUrl` to `misc/drupal.js`, and so because caching and JS aggregation is enabled on Farmier hosting that function would not be available in the cached JS files. Clearing the cache fixed it.
